llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -8933,14 +8933,15 @@ void LoopVectorizationPlanner::buildVPlansWithVPRecipes(ElementCount MinVF,
   for (ElementCount VF = MinVF; ElementCount::isKnownLT(VF, MaxVFTimes2);) {
     VFRange SubRange = {VF, MaxVFTimes2};
     if (auto Plan = tryToBuildVPlanWithVPRecipes(SubRange)) {
+      bool HasScalarVF = Plan->hasVF(ElementCount::getFixed(1));
```

Comment:
Is it possible to have a plan with both scalar and non scalar VFs? Could we use `Plan->hasScalarVFOnly()`

Reply:
I think there's nothing that completely rules it out at the moment, but I couldn't find a test case, although some of the unit tests did have such plans. I'll look a bit more, but for now probably best to be conservative and keep it as is.